public/scripts/main.js

```diff
@@ -366,68 +371,104 @@ let createImgElements = function(imgObj) {
 //  image.style.maxHeight = '80%';
   containerDiv.appendChild(image);
   return containerDiv;
-}
-
-var divModalContainer = document.querySelector('div.modal-content');
-var closeModal = document.querySelector('div.modal-footer>a')
-var addImageModal = document.querySelector('#add-image');
-
-//############### ADD IMG MODAL ##############################
-
-addImageModal.addEventListener('click', function() {
-
-  var divForm = createImageForm();
-  var divButtons = createButtons('image', divForm);
-  divForm.appendChild(divButtons);
-  divModalContainer.appendChild(divForm);
-})
-
-var createImageForm = function() {
-  var divForm = createElementWithClasses('div', ['div-form']);
-  var imageForm = document.createElement('form');
+};
 
+var createImgForm = function() { 
   var input = document.createElement('input');
   input.setAttribute('id', 'url-form');
   input.setAttribute('name', 'imageUrl');
   input.setAttribute('placeholder', 'Enter url here...')
   input.className = 'url-image-form';
+  return input;  
+};
 
-  var titleArea = createTitleInput()
-
-  imageForm.appendChild(titleArea);
-  imageForm.appendChild(input);
-  divForm.appendChild(imageForm);
+//########### ADD IMG MODAL ######################
+var addImageModal = document.querySelector('#add-image');
 
-  return divForm;
-}
+addImageModal.addEventListener('click', function() {
+  var imgElem = createImgForm();
+  var divFormObj = createFormTemplate(imgElem);
+  var objType = 'image';
+  divModalContainer.appendChild(divFormObj.divForm);
+  
+  divFormObj.buttonAdd.addEventListener('click', function() {
+    let definedObj = defineObjType(objType);
+    writeData(definedObj).then(render);
+    deleteDivForm();
+    closeModal.click();
```

Comment:
do we need to get rid off those clicks? Jonathan says it's a bad practice and that it won't work on mobile. But it did work just fine on mobile